File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/ReadPropertyValuesRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;

import java.util.Set;

public class ReadPropertyValuesRequest {
    private final String id;
    private final Set<String> propertyNames;
    private final ObjectServiceContext context;

    public ReadPropertyValuesRequest(
            ObjectServiceContext context,
            Set<String> propertyNames)
    {
        this.id = context.getObjectId().getId();
        this.propertyNames = propertyNames;
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public Set<String> getPropertyNames() {
        return propertyNames;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/ReadPropertyValuesRequest.ts
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class ReadPropertyValuesRequest {
    private readonly _id: string;
    private readonly _propertyNames: Set<string>;
    private readonly _context: ObjectServiceContext;

    constructor(
        context: ObjectServiceContext,
        propertyNames: Set<string>,
    ) {
        this._id = context.objectId.id;
        this._propertyNames = propertyNames;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get propertyNames(): Set<string> {
        return this._propertyNames;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}